File: src/models/pirep.ts
import { Document, model, Schema } from 'mongoose';

interface IPirep extends Document {
	reportTime: Date;
	location: string;
	aircraft: string;
	flightLevel: string;
	skyCond?: string;
	turbulence?: string;
	icing?: string;
	vis?: string;
	temp?: string;
	wind?: string;
	urgent: boolean;
	raw: string;
	manual: boolean;
}

const PirepSchema = new Schema<IPirep>(
	{
		reportTime: { type: Date, required: true },
		location: { type: String, required: true },
		aircraft: { type: String, required: true },
		flightLevel: { type: String, required: true },
		skyCond: { type: String },
		turbulence: { type: String },
		icing: { type: String },
		vis: { type: String },
		temp: { type: String },
		wind: { type: String },
		urgent: { type: Boolean, required: true, default: false },
		raw: { type: String, required: true },
		manual: { type: Boolean, required: true, default: false },
	},
	{ collection: 'pirep' },
);

export const PirepModel = model<IPirep>('pirep', PirepSchema);

File: internal/models/pirep.go
package models

import (
	"time"

	"go.mongodb.org/mongo-driver/v2/bson"
)

type Pirep struct {
	ID            bson.ObjectID `bson:"_id,omitempty"`
	ReportTime    time.Time     `bson:"reportTime"`
	Location      string        `bson:"location"`
	Aircraft      string        `bson:"aircraft"`
	FlightLevel   string        `bson:"flightLevel"`
	SkyConditions *string       `bson:"skyCond"`
	Turbulence    *string       `bson:"turbulence"`
	Icing         *string       `bson:"icing"`
	Visibility    *string       `bson:"vis"`
	Temperatrue   *string       `bson:"temp"`
	Wind          *string       `bson:"wind"`
	Urgent        bool          `bson:"urgent"`
	Raw           string        `bson:"raw"`
	Manual        bool          `bson:"manual"`
}
